l2geth/miner/worker.go

```diff
@@ -567,18 +568,26 @@ func (w *worker) batchStartLoop() {
 						// Split the pending transactions into locals and remotes
 						localTxs, remoteTxs := make(map[common.Address]types.Transactions), pending
 						// TODO mev
-						var txsQueue types.Transactions
+						var rawTxsQueue types.Transactions
 						for _, account := range w.eth.TxPool().Locals() {
 							if txs := remoteTxs[account]; len(txs) > 0 {
 								delete(remoteTxs, account)
 								localTxs[account] = txs
-								txsQueue = append(txsQueue, txs...)
+								rawTxsQueue = append(rawTxsQueue, txs...)
 							}
 						}
 						for _, txs := range remoteTxs {
-							txsQueue = append(txsQueue, txs...)
+							rawTxsQueue = append(rawTxsQueue, txs...)
 						}
-
+						var txsQueue types.Transactions
+						for _, tx := range rawTxsQueue {
+							if err := w.eth.SyncService().VerifyFee(tx); err == nil {
```

Comment:
Do `VerifyFee` again here. I think the `updateGasPriceTx` won't be too frequent.  Maybe is most batches, there is no `updateGasPriceTx`. So I suggest to add a map to save the gasPrice value which the tx has been checked against. If the previous gasPrice value is less than now, then redo  `VerifyFee`, otherwise, just skip.